src/App.tsx

```diff
@@ -1,29 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const DEFAULT_PAGE = 1;
+const items: string[] = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(DEFAULT_PAGE);
+
+  const startItem = (currentPage - 1) * itemsPerPage;
+  let endItem = startItem + itemsPerPage;
+
+  if (endItem > items.length) {
+    endItem = items.length;
+  }
+
+  const handlerPageChange = (value: number) => {
+    setCurrentPage(value);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${startItem + 1} - ${endItem} of ${items.length})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            value={itemsPerPage}
+            onChange={event => {
+              setItemsPerPage(Number(event.target.value));
+              setCurrentPage(DEFAULT_PAGE);
+            }}
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**